pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -231,13 +258,21 @@ func (s *kafkaSink) EmitRow(
 		Metadata: messageMetadata{alloc: alloc, mvcc: mvcc, updateMetrics: s.metrics.recordOneMessage()},
 	}
 	s.stats.startMessage(int64(msg.Key.Length() + msg.Value.Length()))
-	return s.emitMessage(ctx, msg)
+	return s.maybeMarkFatal(s.emitMessage(ctx, msg))
```

Comment:
I don't see the implementation of this method, but I can guess... And i'm worried.
I suppose it's perfectly possible that a blip happens when feed runs w/out batching, and you fail to emit the message.  We probably don't want to mark feed failed.

I thought that at some point we will be relying on a "sane-er" retry behavior, where coordinator notices that it has retried many times, but the water mark does not advance.

Reply:
Implementation is lower down in the same file and it should address your concern (but take a look!) It only marks fatal when it's the specific Kafka message too large error and batching is disabled.